### PR TITLE
feat-delete and toggle and search

### PR DESCRIPTION
**Pull request by gurtatiLND**
# What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Styling
- [ ] Build
- [ ] Chore

## Description

- **Replaced Drag-and-Drop Library**:
  - Removed `@dnd-kit`.
  - Installed and integrated `@hello-pangea/dnd`.
- **Implemented Toggle and Delete Functions**:
  - Added `toggle` and `delete` functionality to each `SortableItem`.
  - Integrated click handlers for toggling an item's state and deleting items.
- **Added `deleteFromDb` Function in `DatabaseManager`**
- **Refactored Search Logic**:
  - Removed the old `SearchBar` component.
  - Updated the `Search` component to receive props from the `Toolkit` component and return filtered data to the parent component.
- **Removed Old `ToolkitHeader`**  
- **Add DragAndDrop icon to `SortableItem`**

## Screenshots

Screenshot 2024-12-09 at 11 21 50

## UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_

- [ ] Semantic HTML implemented?
- [ ] Checked with [axe DevTools](https://www.deque.com/axe/) and addressed `Critical` and `Serious` issues?
- [ ] Color contrast tested?

## Added/updated tests?

_Please aim to keep the code coverage percentage at 80% and above._

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## Delete branch after merge?

- [x] Yes
- [ ] No

## What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,10 +12,10 @@
     "test:watch": "jest --watch"
   },
   "dependencies": {
-    "@dnd-kit/core": "^6.2.0",
-    "@dnd-kit/sortable": "^9.0.0",
+    "@hello-pangea/dnd": "^17.0.0",
     "@heroicons/react": "^2.2.0",
     "@serwist/next": "^9.0.10",
+    "@testing-library/user-event": "^14.5.2",
```
Comment (maxitect) [inline package.json:18]: I think this is a dev dependency, could you uninstall and reinstall as dev please? Sowrry!

File: src/app/toolkit/page.tsx [hunk 2]
```diff
@@ -1,6 +1,5 @@
 "use client";
```
Comment (maxitect) [inline src/app/toolkit/page.tsx:1]:
```suggestion
```

File: src/ui/layout/ToolkitHeader.tsx
Comment (maxitect) [inline src/ui/layout/ToolkitHeader.tsx:1]: This should be deleted (it is no longer in main)

File: src/ui/shared/Search.tsx [hunk 3]
```diff
@@ -1,67 +1,135 @@
...
+//               stroke="currentColor"
+//               strokeWidth={2}
+//               strokeLinecap="round"
+//             />
+//           </svg>
+//         </button>
+//       )}
+//     </div>
+//   );
+
```
Comment (maxitect) [inline src/ui/shared/Search.tsx:135]: Possible to delete this? If so commit below
```suggestion
```

**Review by maxitect — CHANGES_REQUESTED** [review]
Thanks Tanya the features here looks really great and lots of work, absolutely fantastico!

Main change needs to re-implement Jason's category filtering feature from PR #51 (as discussed already)

Then a couple of minor tweaks suggested in comments.

File: package.json (hunk 1 above)
Comment (gurtatiLND) [reply package.json]: Done! Thank you for pointing that out

File: src/app/toolkit/page.tsx (hunk 2 above)
Comment (gurtatiLND) [reply src/app/toolkit/page.tsx:1]: done as well!

File: src/ui/layout/ToolkitHeader.tsx
Comment (gurtatiLND) [reply src/ui/layout/ToolkitHeader.tsx:1]: looks like it was deleted. thank you!

File: src/ui/shared/Search.tsx (hunk 3 above)
Comment (gurtatiLND) [reply src/ui/shared/Search.tsx]: deleted

**Review by maxitect — CHANGES_REQUESTED** [review]
Thanks for making all those changes! 

Sorry though - the category filtering is still not working as in main, when you select a category filter and go back to All, it is not deleting the filter and displaying all the tools

**Pull request merged**